clippy_lints/src/casts/cast_possible_truncation.rs

```diff
@@ -163,12 +172,15 @@ pub(super) fn check(
         _ => return,
     };
 
-    let name_of_cast_from = snippet(cx, cast_expr.span, "..");
     let cast_to_snip = snippet(cx, cast_to_span, "..");
-    let suggestion = format!("{cast_to_snip}::try_from({name_of_cast_from})");
+    let suggestion = if cast_to_snip == "_" {
+        format!("{}.try_into()", Sugg::hir(cx, cast_expr, "..").maybe_par())
+    } else {
+        format!("{cast_to_snip}::try_from({})", Sugg::hir(cx, cast_expr, ".."))
+    };
 
     span_lint_and_then(cx, CAST_POSSIBLE_TRUNCATION, expr.span, &msg, |diag| {
-        diag.help("if this is intentional allow the lint with `#[allow(clippy::cast_possible_truncation)]` ...");
+        diag.help(help);
         diag.span_suggestion_with_style(
             expr.span,
             "... or use `try_from` and handle the error accordingly",
```

Comment:
Rather than returning early let's add an `if cast_to.is_floating_point() { ... }` around the suggestion creation here, that way we also catch the `f64` -> `f32` case

Reply:
@Alexendoo  thanks for reviewing btw~

 I somehow thought float to float conversion impls the TryFrom trait but I was wrong :man_facepalming:  also I assume you meant `if !cast_from.is_floating_point()` right?